pennylane/workflow/qnode.py

```diff
@@ -207,15 +207,26 @@ def _to_qfunc_output_type(
     if has_partitioned_shots:
         return tuple(_to_qfunc_output_type(r, qfunc_output, False) for r in results)
 
-    # Special case of single Measurement in a list
-    if isinstance(qfunc_output, Sequence) and len(qfunc_output) == 1:
-        results = (results,)
+    qfunc_output_leaves, qfunc_output_structure = qml.pytrees.flatten(
+        qfunc_output, is_leaf=lambda obj: isinstance(obj, (qml.measurements.MeasurementProcess))
+    )
+
+    results_leaves = qml.pytrees.flatten(results, is_leaf=lambda obj: isinstance(obj, dict))[0]
+    print(qfunc_output)
+    print(results)
 
-    # If the return type is not tuple (list or ndarray) (Autograd and TF backprop removed)
-    if isinstance(qfunc_output, (tuple, qml.measurements.MeasurementProcess)):
+    if len(results_leaves) != len(qfunc_output_leaves):
+        if qml.math.get_interface(qfunc_output) in {"numpy", "autograd"}:
+            return type(qfunc_output)(results)
         return results

```

Comment:
```suggestion
        if isinstance(qfunc_output, (Sequence, qml.measurements.MeasurementProcess)):
            return results
        return type(qfunc_output)(results)
```
this should fix things 🤞

Reply:
`qml.expval(qml.Z(0))` has an "autograd" interface 🤦‍♀️